lua/neogit/config.lua

```diff
@@ -1,157 +1,457 @@
-  },
-  mappings = {
-    finder = {
-      ["<cr>"] = "Select",
-      ["<c-c>"] = "Close",
-      ["<esc>"] = "Close",
-      ["<c-n>"] = "Next",
-      ["<c-p>"] = "Previous",
-      ["<down>"] = "Next",
-      ["<up>"] = "Previous",
-      ["<tab>"] = "MultiselectToggleNext",
-      ["<s-tab>"] = "MultiselectTogglePrevious",
-      ["<c-j>"] = "NOP",
+    preview_buffer = {
+      kind = "split",
     },
-    status = {
-      ["q"] = "Close",
-      ["I"] = "InitRepo",
-      ["1"] = "Depth1",
-      ["2"] = "Depth2",
-      ["3"] = "Depth3",
-      ["4"] = "Depth4",
-      ["<tab>"] = "Toggle",
-      ["x"] = "Discard",
-      ["s"] = "Stage",
-      ["S"] = "StageUnstaged",
-      ["<c-s>"] = "StageAll",
-      ["u"] = "Unstage",
-      ["U"] = "UnstageStaged",
-      ["d"] = "DiffAtFile",
-      ["$"] = "CommandHistory",
-      ["#"] = "Console",
-      ["<c-r>"] = "RefreshBuffer",
-      ["<enter>"] = "GoToFile",
-      ["<c-v>"] = "VSplitOpen",
-      ["<c-x>"] = "SplitOpen",
-      ["<c-t>"] = "TabOpen",
-      ["?"] = "HelpPopup",
-      ["D"] = "DiffPopup",
-      ["p"] = "PullPopup",
-      ["r"] = "RebasePopup",
-      ["m"] = "MergePopup",
-      ["P"] = "PushPopup",
-      ["c"] = "CommitPopup",
-      ["L"] = "LogPopup",
-      ["_"] = "RevertPopup",
-      ["Z"] = "StashPopup",
-      ["A"] = "CherryPickPopup",
-      ["b"] = "BranchPopup",
-      ["f"] = "FetchPopup",
-      ["X"] = "ResetPopup",
-      ["M"] = "RemotePopup",
-      ["{"] = "GoToPreviousHunkHeader",
-      ["}"] = "GoToNextHunkHeader",
+    popup = {
+      kind = "split",
+    },
+    signs = {
+      hunk = { "", "" },
+      item = { ">", "v" },
+      section = { ">", "v" },
     },
-  },
-}
+    integrations = setmetatable({}, {
+      __index = function(_, key)
+        local ok, value = pcall(require, key)
+        return ok and value or false
+      end,
+    }),
+    sections = {
+      untracked = {
+        folded = false,
+      },
+      unstaged = {
+        folded = false,
+      },
+      staged = {
+        folded = false,
+      },
+      stashes = {
+        folded = true,
+      },
+      unpulled = {
+        folded = true,
+      },
+      unmerged = {
+        folded = false,
+      },
+      recent = {
+        folded = true,
+      },
+      rebase = {
+        folded = true,
+      },
+    },
+    ignored_settings = {
+      "NeogitPushPopup--force-with-lease",
+      "NeogitPushPopup--force",
+      "NeogitCommitPopup--allow-empty",
+      "NeogitRevertPopup--no-edit", -- TODO: Fix incompatible switches with default enables
+    },
+    mappings = {
+      finder = {
+        ["Select"] = { "<cr>" },
+        ["Close"] = { "<c-c>", "<esc>" },
+        ["Next"] = { "<c-n>", "<down>" },
+        ["Previous"] = { "<c-p>", "<up>" },
+        ["MultiselectToggleNext"] = { "<tab>" },
+        ["MultiselectTogglePrevious"] = { "<s-tab>" },
+        ["NOP"] = { "<c-j>" },
+      },
+      status = {
```

Comment:
Setting to false doesn't work for the reverse lookup unless we do some preparsing to remove the key from the table entirely.

The primary issue comes when they have multiple mappings set to a command, in the reverse lookups we'll still fail to get the reversed map, for instance setup like so:

```lua
neogit.setup({
    mappings = {
        status = {
            ["s"] = "Status",
            ["b"] = "Status"
        }
    }
})
```

That will error, and so will setting them to false. We could check if a command has multiple maps and reject it in the validation, but we still don't have a way of disabling the mapping unless we do some preprocessing to remove keys that are set to false prior to getting the reverse map.

Furthermore, a user no longer has to disable default maps with the `command -> key` setup, by setting their new mapping they necessarily disable the default map.

I can do the preprocessing and whatnot if you'd prefer to keep it `key -> command`, but I think there's less extra code and processing done with `command -> key`. Up to you, just let me know.

Reply:
I see. Given my understadning, the reverse lookup is mainly for user displaying purposes, and not functionality related?

Just throwing things at the wall to see what sticks and what doesn't

https://github.com/NeogitOrg/neogit/pull/659

Would that one solve the reverse lookup missing functionality?

Reply:
It solves the reverse lookup presentation, but not the binding issues. We'd still have to do some preprocessing :/.

Reply:
Yes unfortunately, We'd need a `filter` or just an `if` in that for loop as well.

How does the unbinding work with your changes?

Reply:
Well you're directly setting the keybinds for the maps like so:

```lua
["Status"] = "s"
```

To unbind `Status` entirely you'd just do:

```lua
["Status"] = ""
```

And since the commands are the key in my version there's not really a need to do a reversed map as far as I know, avoiding the whole duplicate key stuff we have right now.

Reply:
So, while I do see the issues with having `["s"] = "Stage"`, I do think that doing the "expected" thing has a lot of value, which in this case is to stick to the current mapping schema. Also, for the longest time, I've had both `h` and `l` bound to `Toggle`, to treat folds like a file-tree, going "up" a level and "down" a level. 

I'm not very concerned about users unbinding keys.. Honestly, I think of neogit as an app within an app, so, while it's "in" nvim, overloading common things like "v" is fair game. 

I think the most compelling reason to adopt this change is to use the *correct* bindings in all locations. The log view, commit view, and reflog view can all just use `config.mapping.cherry_pick` and get the right thing.

So, I'm in favor (mostly) for that reason.

Reply:
If y'all don't think the mapping changes here are worth it, then what I can do is extract all the config validation and tests to a separate PR and we can look at that alone there.

In terms of the mappings, if we don't want to go with my changes I guess we can do this:

If a user sets a default mapping to false then we remove the key entirely from the table. Secondly, if they have the same command mapped to multiple keys we can do a.) fail validation or b.) rework some of the internals to better handle these multiple mappings (such as not using reversed tables and instead searching for all commands(?)).

Reply:
Yes I am not sure either, which is why I preemptively opened a PR (#659) that reworks the internals for multi mappings which makes the hints and everything work. I know @CKolkey uses two mappings for one action, and I think others may be as well, such as `=` for staging. If possible, it would thus be good to keep it supported, removing functionality is always a bit iffy, especially if there is no workaround.

I have absolutely no objections against mucking about in the internals to make it work though.

Reply:
I think you're patch over there at #659 is what we should use instead of this as it preserves functionality and addresses all of the bugs known in other issues. I already have a branch based on it for the config validation.